treeherder/webapp/api/performance_data.py

```diff
@@ -743,6 +748,250 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
+                    ),
+                    'extra_options': base_result.get('extra_options', ''),
+                    'base_repository_name': base_repository_name,
+                    'new_repository_name': new_repository_name,
+                    'base_measurement_unit': base_result.get('measurement_unit', ''),
+                    'new_measurement_unit': new_result.get('measurement_unit', ''),
+                    'base_runs': sorted(base_values),
+                    'new_runs': sorted(new_values),
+                    'base_avg_value': base_avg_value,
+                    'new_avg_value': new_avg_value,
+                    'base_stddev': base_stddev,
+                    'new_stddev': new_stddev,
+                    'base_stddev_pct': base_stddev_pct,
+                    'new_stddev_pct': new_stddev_pct,
+                    'base_retriggerable_job_ids': base_grouped_job_ids.get(
+                        base_result.get('id', ''), []
+                    ),
+                    'new_retriggerable_job_ids': new_grouped_job_ids.get(
+                        new_result.get('id', ''), []
+                    ),
+                }
+
+                self.queryset.append(row_result)
+
+        serializer = self.get_serializer(self.queryset, many=True)
+        serialized_data = serializer.data
+
+        return Response(data=serialized_data)
+
+    def get_option_collection_map(self):
+        option_collection = OptionCollection.objects.select_related('option').values(
+            'id', 'option__name'
+        )
+        option_collection_map = {
+            item['id']: item['option__name'] for item in list(option_collection)
+        }
+        return option_collection_map
+
+    def get_avg_and_stddev(self, base_values, new_values, header):
+        if header == self.noise_metric_header:
+            base_stddev = 1
+            new_stddev = 1
+            base_avg_value = sqrt(
+                functools.reduce(lambda a, b: a + b, map(lambda x: x**2, base_values))
+            )
+            new_avg_value = sqrt(
+                functools.reduce(lambda a, b: a + b, map(lambda x: x**2, new_values))
+            )
+        else:
+            base_avg_value = mean(base_values) if len(base_values) else 0
+            new_avg_value = mean(new_values) if len(new_values) else 0
+            base_stddev = stdev(base_values) if len(base_values) >= 2 else None
+            new_stddev = stdev(new_values) if len(new_values) >= 2 else None
+        return base_avg_value, base_stddev, new_avg_value, new_stddev
+
+    def get_percentage(self, part, whole):
+        percentage = 0
+        if whole:
+            percentage = (100 * part) / whole
+        return percentage
+
+    def get_grouped_perf_data(self, perf_data):
+        grouped_values = defaultdict(list)
+        grouped_job_ids = defaultdict(list)
+        for signature_id, value, job_id in perf_data.values_list('signature_id', 'value', 'job_id'):
+            if value is not None:
```

Comment:
`if value:` ?

Reply:
This endpoint is based on the `performance/summary/` endpoint, and I would like to keep this logic as the initial one.